### PR TITLE
Migrate to new username system

### PR DESCRIPTION
**Pull request by Nuckyz**

### CODE REVIEW

File: lib/structures/User.ts [hunk 1]
```diff
@@ -62,9 +68,9 @@ export default class User extends Base {
         }
     }
 
-    /** The default avatar value of this user (discriminator modulo 5). */
+    /** The default avatar value of this user (user id shifted by 22, then modulo 5). */
```
Comment (DonovanDMC) [inline lib/structures/User.ts]:
```suggestion
    /** The default avatar value of this user (user id right shifted by 22, then modulo 5). */
```

**Comment by DonovanDMC** [comment]
We also need to figure out how we're going to deal with this (not adding webhooks to the cache)
https://github.com/OceanicJS/Oceanic/blob/be781c34c6dc0f20a441c5ec7a71c4c9e6cbd61d/lib/structures/Message.ts#L148

File: lib/structures/User.ts [hunk 2]
```diff
@@ -62,9 +68,9 @@ export default class User extends Base {
         }
     }
 
-    /** The default avatar value of this user (discriminator modulo 5). */
+    /** The default avatar value of this user (user id right shifted by 22, then modulo 5). */
     get defaultAvatar(): number {
-        return Number(this.discriminator) % 5;
+        return Number(BigInt(this.id) >> 22n) % 5;
     }
```
Comment (A5rocks) [inline lib/structures/User.ts]: https://github.com/discord/discord-api-docs/pull/6130#issuecomment-1556391316 FYI

**Comment by Syjalo** [comment]
> We also need to figure out how we're going to deal with this (not adding webhooks to the cache)
> 
> https://github.com/OceanicJS/Oceanic/blob/be781c34c6dc0f20a441c5ec7a71c4c9e6cbd61d/lib/structures/Message.ts#L148

There's `data.webhook_id`

**Comment by DonovanDMC** [comment]
force push to master

**Pull request merged**